### PR TITLE
Fix for loops

### PR DESCRIPTION
**Pull request by pianoslum**
So.. the same thing for stdlib.
I tried to use map/filter where possible.

When the line was very long I wasn't sure how to indent...

### CODE REVIEW

File: msgHdrUtils.js [hunk 1]
```diff
@@ -215,18 +210,16 @@ function msgHdrGetTags (aMsgHdr) {
...
   let newTagList = aTags;
-  for each (let [, tag] in Iterator(newTagList))
+  for (let tag of newTagList)
     newTags[tag.key] = null;
 
-  let toAdd = [x.key for each ([, x] in Iterator(newTagList))
-    if (!(x.key in oldTags))];
-  let toRemove = [x.key for each ([, x] in Iterator(oldTagList))
-    if (!(x.key in newTags))];
+  let toAdd = newTagList.map(x => x.key);
```
Comment (protz) [inline msgHdrUtils.js]: You're missing a call to `filter` here.

**Comment by protz** [comment]
Thanks! That's awesome. I'll merge once you fix the tidbit above.

**Comment by pianoslum** [comment]
Ok, this should be it!

**Comment by protz** [comment]
Fantastic. Feel free to submit, along with your next pull request to the conversations repository, the bit that updates the git submodule in there.

**Pull request merged**